src/CyBy/Draw/Internal/Graph.idr

```diff
@@ -604,6 +604,34 @@ newEdges t ps = do
   (a1,a2) <- ps
   (\(x,l) => (a1,x,l)) <$> neighboursAsPairs t a2
 
+createStableBond :
+     {k,m : _}
+  -> List (Fin k, Fin m)
+  -> Edge m CDBond
+  -> Maybe $ Edge k CDBond
+createStableBond nsToMerge bnd =
+  let filter := foldMap {m=List (Fin k, Fin m)}
+                  (\(nm,nt) => if nt == bnd.node1 || nt == bnd.node2
+                                 then [(nm, nt)]
+                                 else [])
+                  nsToMerge
+   in case filter of
+        [(nm',_),(nm'',_)] => mkEdge nm' nm'' bnd.label
+        _                  => Nothing
+
+-- filters out bonds of the template, where both nodes are merged and therefore
+-- the bond would be deleted
+-- transfers this bonds to the nodes of the origin molecule, therefore the bonds
+-- are not deleted and are 'stable'
+stableBonds :
+     {k,m : _}
+  -> CDIGraph m
+  -> (nsToMerge : List (Fin k, Fin m))
+  -> List $ Edge k CDBond
+stableBonds t nsToMerge =
+  let tempEdges := edges t
+   in mapMaybe (createStableBond nsToMerge) tempEdges
```

Comment:
Personally, I think this is so small, it could just go into the `let` expression of `mergeGraphs'`. Also, don't introduce a variable that's only used once (`tempEdges`) unless  the remaining expression gets too long or is unclear otherwise. `tempEdges` does not add clarity over `(edges t)`.